contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -728,9 +822,11 @@ def all_tests_successful(self) -> bool:
         :returns: bool where True indicates all tests succeeded (they existed, complete and were
             PASS/SKIP)
         """
-        # If no tests are defined, we consider it a failure for the detection
+        # If no tests are defined, we consider it a success for the detection (this detection was
+        # skipped for testing). Note that the existence of at least one test is enforced by Pydantic
+        # validation already, with a few specific exceptions
         if len(self.tests) == 0:
-            return False
+            return True
```

Comment:
Notable change. I think this fine, as we are enforcing the existence of at least one test where it matters (at model build, for all but a handful of edge cases)

Reply:
Yeah, I agree. I will close this out, but this is where I think we COULD do with things like defining different Classes for each `type:` or `status:`. That's a huge rework, though, and I think this is a good change for what we have.
The comment above this statement  
```
        # If no tests are defined, we consider it a success for the detection (this detection was
        # skipped for testing). Note that the existence of at least one test is enforced by Pydantic
        # validation already, with a few specific exceptions
```
Also gives good insight into the logic

Reply:
Can you elaborate a little on what you mean by different Classes?

Reply:
I mean that a Production Detection, an Experimental Detection, and a Deprecated Detection all have different validations/requirements.
Similarly, TTP, Anomaly, Hunting, and Correlation may have different requirements and behaviors.  
Having all these requirements and validations and behaviors expressed via complicated IF/ELSE logic is not as clean as making `Detection` an abstract class (and making Production/Experimental/Deprecated and/or TTP/Anomaly/Hunting/Correlation non-abstract classes that define behavior specific to their types).

Reply:
Ahhhh I see. I like that idea a lot! And I think it makes a lot of sense to compartmentalize validation requirements like that

Reply:
Leaving this unresolved intentionally to look back to later (not blocking this PR)